### PR TITLE
:hammer: change signatures

### PR DESCRIPTION
**Pull request by fabienjuif**
#35 

README : https://github.com/Trampss/trampss-redux-factory/blob/signature/README.md

TYPES : https://github.com/Trampss/trampss-redux-factory/blob/signature/TYPES.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,8 +16,8 @@ Factory of Redux reducers and their associated actions and selectors.
 `trampss-redux-factory` creates generic reducers, actions and selectors in two lines.
 
 ```es6
-import factory from 'trampss-redux-factory'
-export default factory(/* middlewares */)('id')('api')('todos')
+import { map } from 'trampss-redux-factory'
+export default map({ key: 'id', path: 'api', name: 'todos' })
 ```
 That's it, you exported a reducer function you can register thanks to combinerReducer in Redux.
```
Comment (bpetetot) [inline README.md]: "That's it, you just have to export the reducer function and you can register it through combinerReducer in Redux"

File: src/index.js [hunk 2]
```diff
@@ -3,33 +3,56 @@ import * as selectors from './selectors'
...
+    if (typeof params === 'string') return getWrappedStore()({ name: params, ...forcedOptions })
+    return getWrappedStore()({ ...params, ...forcedOptions })
   }
 
-  return func(middlewares)(key)(path)(types[type])(prefix)(name)
+  // not a valid param
+  error()
 }
+
+export const uniq = factory({ type: 'uniq' })
```
Comment (guillaumecrespel) [inline src/index.js]: uniq is not connoted with the function uniq of lodash :/
`uniq` => `single` ?
`map` => `multi` ?

i think, the name of funciton dont match with the features.

File: README.md (hunk 1 above)
Comment (fabienjuif) [inline README.md]: What about `That's it, you just exported the reducer function and now you can register it through combinerReducer in Redux.` ?

The reducer function is already exported (this is the code above the sentence)

File: src/index.js [hunk 3]
```diff
@@ -3,33 +3,56 @@ import * as selectors from './selectors'
...
+      .map(k => ({ [k]: selectors[k](path)(name) })),
+  )
+}
+
+// error :( - not a middleware nor an option parameter
+const error = () => { throw Error('parameter is not a middleware configuration, nor a factory option object.') }
+
+// params checkers
+const isMiddleware = params => params.engine || params.pre || params.post
+const isOptions = params => (!!params.name) || (typeof params === 'string')
```
Comment (fabienjuif) [inline src/index.js]: why ? `!!` is about converting a truthy value to `true`

File: src/index.js (hunk 2 above)
Comment (fabienjuif) [inline src/index.js]: I agree and I would prefer an issue to argue about it

Comment (guillaumecrespel) [inline src/index.js]: `uniq` => `simple` ?
`map` => `complex` ?
@bpetetot An idea ? :)

Comment (fabienjuif) [inline src/index.js]: https://github.com/Trampss/trampss-redux-factory/issues/45

**Pull request merged**